### PR TITLE
Syncing staging <> master branches

### PR DESCRIPTION
**Pull request by PatrickBue**
### Description
Update master.

### Checklist:
- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [x] I have updated the documentation accordingly.
- [n/a] This PR is being made to `staging` and not `master`

### CODE REVIEW

File: contrib/action_recognition/r2p1d/README.md [hunk 1]
```diff
@@ -10,12 +10,12 @@ This project provides utility scripts to fine-tune the model and examples notebo
...
 | [00_webcam](00_webcam.ipynb) | A real-time inference example on Webcam stream |
 | [01_training_introduction](01_training_introduction.ipynb) | An example of training R(2+1)D model on HMDB-51 dataset |
-| [02_video_transformation](02_video_transformation.ipynb) | Examples of video transformations | 
+| [02_video_transformation](02_video_transformation.ipynb) | Examples of video transformations |
 
 Specifically, we use the model pre-trained on 65 million social media videos (IG) presented in "[Large-scale weakly-supervised pre-training for video action recognition (2019)](https://arxiv.org/abs/1905.00561)" paper.
 
 *Note: The official pretrained model weights can be found from [https://github.com/facebookresearch/vmz](https://github.com/facebookresearch/vmz) which are based on caffe2.
-In this repository, we use PyTorch-converted weights from [https://github.com/moabitcoin/ig65m-pytorch](https://github.com/moabitcoin/ig65m-pytorch).*
+In this repository, we use code and the PyTorch-converted weights from [https://github.com/moabitcoin/ig65m-pytorch](https://github.com/moabitcoin/ig65m-pytorch).*
```
Comment (loomlike) [inline contrib/action_recognition/r2p1d/README.md]: The original statement is correct. We use the model weights only, not the codes. All the codes we wrote except some video transformations which are from Torch official.

The model weights, we get from torchHub by using torch api as instructed in the github we linked (the author's github how uploaded the weight to torchHub).
```
model = torch.hub.load(
    TORCH_R2PLUS1D, model_name, num_classes=MODELS[model_name], pretrained=True
)
```

File: scenarios/action_recognition/README.md [hunk 2]
```diff
@@ -1,66 +1,111 @@
...
+Action recognition is an active field of research, with large number of approaches being published every year. One of the approaches which stands out is the  **R(2+1)D** model which is described in the 2019 paper "[Large-scale weakly-supervised pre-training for video action recognition](https://arxiv.org/abs/1905.00561)".
 
+R(2+1)D is highly accurate and at the same time significantly faster than other approaches:
+- Its accuracy comes in large parts from an extra pre-training step which uses 65 million automatically annotated video clips.
+- Its speed comes from simply using video frames as input. Many other state-of-the-art methods require optical flow fields to be pre-computed which is computationally expensive (see the "Inference speed" section below).
 
-## Installation
-1. Setup conda environment
-`conda env create -f environment.yml`
+We base our implementation on this [github](https://github.com/moabitcoin/ig65m-pytorch) repository, with added functionality to make training and evaluating custom models more user-friendly. We were able to re-produce the published accuracy for the HMDB-51 dataset as can be seen in this table:
```
Comment (loomlike) [inline scenarios/action_recognition/README.md]: Again, we don't borrow any codes from that repo. We use the pretrained model weight from the repo (which the author uploaded to torchhub).

File: scenarios/action_recognition/README.md [hunk 3]
```diff
@@ -1,66 +1,111 @@
-# Human Action Recognition 
+# Human Action Recognition
+
 ```diff
-+ 12/4/2019: This is work-in-progress.
++ Feb 2020: We are working on moving all code from contrib\action_recognition to this folder.
++           While this work is ongoing, please visit both locations for implementations and documentation.
```
Comment (loomlike) [inline scenarios/action_recognition/README.md]: I believe 'master' branch should only include product-ready / fully functional and fully tested codes (sometimes things can be broken in master though). In that sense, I think we should remove any components that are still work-in-progress from master. Users who want to taste future features always can checkout to staging - or use contrib (that's why we add contrib) - and use them at their own risks.

File: contrib/action_recognition/r2p1d/README.md (hunk 1 above)
Comment (PatrickBue) [reply contrib/action_recognition/r2p1d/README.md]: Thanks for the clarification. I'll have a few related questions to this once we meet.

File: scenarios/action_recognition/README.md (hunk 2 above)
Comment (PatrickBue) [reply scenarios/action_recognition/README.md]: Thanks for the clarification. I'll have a few related questions to this once we meet.

**Pull request merged**